homeassistant/components/subaru/lock.py

```diff
@@ -0,0 +1,92 @@
+"""Support for Subaru door locks."""
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.lock import LockEntity
+from homeassistant.const import SERVICE_LOCK, SERVICE_UNLOCK
+from homeassistant.helpers import config_validation as cv, entity_platform
+
+from . import DOMAIN, get_device_info
+from .const import (
+    ATTR_DOOR,
+    ENTRY_CONTROLLER,
+    ENTRY_VEHICLES,
+    SERVICE_UNLOCK_SPECIFIC_DOOR,
+    UNLOCK_DOOR_ALL,
+    UNLOCK_VALID_DOORS,
+    VEHICLE_HAS_REMOTE_SERVICE,
+    VEHICLE_NAME,
+    VEHICLE_VIN,
+)
+from .remote_service import async_call_remote_service
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the Subaru locks by config_entry."""
+    entry = hass.data[DOMAIN][config_entry.entry_id]
+    controller = entry[ENTRY_CONTROLLER]
+    vehicle_info = entry[ENTRY_VEHICLES]
+    async_add_entities(
+        SubaruLock(vehicle, controller)
+        for vehicle in vehicle_info.values()
+        if vehicle[VEHICLE_HAS_REMOTE_SERVICE]
+    )
+
+    platform = entity_platform.async_get_current_platform()
+
+    platform.async_register_entity_service(
+        SERVICE_UNLOCK_SPECIFIC_DOOR,
+        {vol.Required(ATTR_DOOR): cv.string},
```

Comment:
You can use `vol.In` to limit this to valid values.

Reply:
Change incorporated.